### DISCUSSION TITLE
Is there an example of ble_bonding_central.py like ble_bonding_peripheral.py?The central I wrote myself never received the _IRQ_PASSKEY_ACTION event when pairing .

### DISCUSSION THREAD

**Discussion by shushi0123** [micropython/micropython#14478] [General]
I have two boards for the esp32-c3, one as a peripheral to bind and pair.I was able to use the ios nrf connect app to connect to the peripheral, and a pop-up window reminded me to enter the password.
However, when I used the other board as the central device, I never received the _IRQ_PASSKEY_ACTION event, so I could not enter the known password. Finally I will receive the _IRQ_ENCRYPTION_UPDATE event, but authenticated flag is 0.

code:
[ble_peripheral.zip](https://github.com/micropython/micropython/files/15301682/ble_peripheral.zip)

import ble_peripheral as BLE
BLE.Setup() #run peripheral

![图片](https://github.com/micropython/micropython/assets/17306722/0ff7768b-ab9f-4559-a47a-56b56c12e00a)
peripheral running states.

**Comment by NENightElves** [comment]
Have you solved the problem yet? I meet the same one.
I have opened an issue https://github.com/micropython/micropython/issues/17805#issue-3282613491.
